openapi/openapi/src/test/scala/endpoints4s/openapi/EndpointsTest.scala

```diff
@@ -143,6 +143,83 @@ class EndpointsTest extends AnyWordSpec with Matchers with OptionValues {
     Fixtures.toSchema(Fixtures.recursiveSchema.docs) shouldBe expectedSchema
   }
 
+  "Recursive expression" in {
+    val expectedSchema =
+      Schema.Reference(
+        "Expression",
+        Some(
+          Schema.OneOf(
+            Schema.EnumeratedAlternatives(
+              Schema.Primitive("integer", Some("int32"), None, None, None) ::
+                Schema.Object(
+                  Schema.Property(
+                    "x",
+                    Schema.Reference("Expression", None, None),
+                    isRequired = true,
+                    description = None
+                  ) :: Schema.Property(
+                    "y",
+                    Schema.Reference("Expression", None, None),
+                    isRequired = true,
+                    description = None
+                  ) :: Nil,
+                  additionalProperties = None,
+                  description = None,
+                  example = None,
+                  title = None
+                ) :: Nil
+            ),
+            None,
+            None,
+            None
+          )
+        ),
+        None
+      )
+    Fixtures.toSchema(Fixtures.expressionSchema.docs) shouldBe expectedSchema
+  }
+  "Mutually recursive types" in {
+    Fixtures.toSchema(Fixtures.mutualRecursiveA.docs) shouldBe Schema.Reference(
```

Comment:
What does `mutualRecursiveB.docs` look like?

Reply:
Here are the schema we got for `Expression` and the `MutualRecursiveA/B`:

```json
{
  "schemas": {
    "Expression": {
      "oneOf": [
        {
          "type": "integer",
          "format": "int32"
        },
        {
          "type": "object",
          "properties": {
            "x": {
              "$ref": "#/components/schemas/Expression"
            },
            "y": {
              "$ref": "#/components/schemas/Expression"
            }
          },
          "required": [
            "x",
            "y"
          ]
        }
      ]
    },
    "A": {
      "type": "object",
      "properties": {
        "b": {
          "$ref": "#/components/schemas/B"
        }
      }
    },
    "B": {
      "type": "object",
      "properties": {
        "a": {
          "$ref": "#/components/schemas/A"
        }
      }
    }
  }
}
```

Reply:
This is very nice!